### PR TITLE
fix: multi publish still loading in the same profile

### PR DESCRIPTION
**Pull request by VanyLaw**
## Description
fix multi publish still loading in the same profile. 
**Reason**: publish with the same profile will used the same project name and profile name to save the history and status. if user click publish again, those value will be messed up.
**Solution**: disable publish button if already publishing with the same profile.

## Task Item

#minor

### CODE REVIEW

File: Composer/packages/client/src/pages/publish/Publish.tsx [hunk 1]
```diff
@@ -86,6 +86,19 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
...
+    () => () => {
+      // if current profile is already publishing, set disable to true
+      if (selectedTargetName && selectedTargetName !== 'all') {
+        if (thisPublishHistory?.length >= 0) {
+          return thisPublishHistory[0]?.status === 202;
+        }
+      } else {
+        return selectedTargetName === 'all';
+      }
+    },
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/pages/publish/Publish.tsx:99]: Why does this need to return a function?

Comment (VanyLaw) [reply Composer/packages/client/src/pages/publish/Publish.tsx:99]: fixed

File: Composer/packages/client/src/pages/publish/Publish.tsx [hunk 2]
```diff
@@ -86,6 +86,17 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
     [projectId, publishTypes]
   );
 
+  const isPublishDisable = useMemo(() => {
+    // if current profile is already publishing, set disable to true
+    if (selectedTargetName && selectedTargetName !== 'all') {
+      if (thisPublishHistory?.length >= 0) {
```
Comment (hatpick) [inline Composer/packages/client/src/pages/publish/Publish.tsx:92]: when this ```if``` is not true, nothing is returned, do you want it to be undefined?

**Comment by hatpick** [comment]
@VanyLaw would you please address the comments?

File: Composer/packages/client/src/pages/publish/Publish.tsx (hunk 2 above)
Comment (VanyLaw) [reply Composer/packages/client/src/pages/publish/Publish.tsx]: fixed

**Comment by VanyLaw** [comment]
> @VanyLaw would you please address the comments?
> Also please update the branch to resolve the conflicts.
@hatpick Thanks for the comment, I addressed it.

**Review by hatpick — APPROVED** [review]

**Comment by VanyLaw** [comment]
Due to publish page was change to a large extent, I need to confirm with @alanlong9278 how to disable those profiles which were used by current publish, make sure it can not select during current publish.
So I close it now. Will send another PR later.

**Pull request closed without merging**